internal/ordered/slice.go

```diff
@@ -0,0 +1,30 @@
+package ordered
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+)
+
+var _ yaml.Unmarshaler = (*Slice)(nil)
+
+// Slice is []any, but unmarshaling into it prefers *Map[string,any] over
+// map[string]any.
+type Slice []any
```

Comment:
given that `ordered.Map` is generic, does it make sense to make `ordered.Slice` generic too? that way we should make the `Steps` field on the `Pipeline` struct a `ordered.Slice[Step]`, which should make using the struct a fair bit easier.

in terms of code rhyming, it makes a bit of sense as well, but it might very well be technically infeasible for whatever reason

Reply:
What I have in the other PR was `Steps`, which fingerprints each one and unmarshals the right kind of `Step`. I didn't want to get into the weeds of defining each kind of `Step` just yet, so I went with `[]any`. It felt a bit off putting something as general as an `ordered.Map`-biased `[]any` parser in the pipeline parser, but since it will be upgraded with specific step types later, that's probably where it belongs.

Reply:
sweet, that makes total sense - i think i might not have realised that `Step` wasn't in this PR yet 😅